node/src/database/test_utils/transaction_wrapper_mock.rs

```diff
@@ -0,0 +1,394 @@
+
+            let mut setup = SetupForBoth {
+                prod_code_calls_conn_used_for_both: false,
+                prod_code_calls_conn: conn,
+                prod_code_calls_transaction_opt: None,
+                requested_preceding_prod_code_calls: 0,
+                stubbed: SetupForStubbed {
+                    stubbed_calls_conn,
+                    stubbed_calls_optional_statements_literals: vec![],
+                },
+            };
+
+            let conn = unsafe { ptr.as_mut().unwrap() };
+            let txn = conn.transaction().unwrap();
+
+            setup.prod_code_calls_transaction_opt = Some(txn);
+
+            Either::Left(setup)
+        };
+
+        Self {
+            calls_counter: RefCell::new(0),
+            setup,
+        }
+    }
+
+    pub fn new_for_only_stubbed(stubbed_calls_conn: Box<dyn ConnectionWrapper>) -> Self {
+        let setup = Either::Right(SetupForStubbed {
+            stubbed_calls_conn,
+            stubbed_calls_optional_statements_literals: vec![],
+        });
+
+        Self {
+            calls_counter: RefCell::new(0),
+            setup,
+        }
+    }
+
+    pub fn count_of_initial_prod_code_calls(mut self, prod_code_calls: usize) -> Self {
+        let dual_setup = self.setup_for_both_or_panic_ref_mut();
+        if dual_setup.requested_preceding_prod_code_calls == 0 {
+            dual_setup.requested_preceding_prod_code_calls = prod_code_calls
+        } else {
+            panic!("Use only single call of \"number_of_prod_code_calls!\"")
+        }
+        self
+    }
+
+    pub fn prod_code_calls_conn_used_for_both(mut self) -> Self {
+        let dual_setup = self.setup_for_both_or_panic_ref_mut();
+        dual_setup.prod_code_calls_conn_used_for_both = true;
+        self
+    }
+
+    pub fn add_single_stubbed_call_from_prod_code_statement(mut self) -> Self {
+        let single_setup = self.setup_for_both_or_panic_ref_mut();
+        single_setup
+            .stubbed
+            .stubbed_calls_optional_statements_literals
+            .push(None);
+        self
+    }
+
+    pub fn add_single_stubbed_call_statement(mut self, statement: &str) -> Self {
+        let single_setup = self.setup_for_both_or_panic_ref_mut();
+        single_setup
+            .stubbed
+            .stubbed_calls_optional_statements_literals
+            .push(Some(statement.to_string()));
+        self
+    }
+
+    fn dual_setup_mismatch() -> ! {
+        panic!(
+            "{}",
+            "This mock was construct as intended for the setup with both prod code and stubbed \
+            calls. Reconsider that step, if you think you will be using only the stubbed calls"
+        )
+    }
+
+    fn setup_for_both_or_panic_ref(&self) -> &SetupForBoth {
+        match self.setup.as_ref() {
+            Either::Left(both) => both,
+            Either::Right(_) => Self::dual_setup_mismatch(),
+        }
+    }
+
+    fn setup_for_both_or_panic_ref_mut(&mut self) -> &mut SetupForBoth {
```

Comment:
This should be renamed to:

```rust
get_setup_for_both_as_mut_or_panic
```

Similarly, This name conveys that
1. It's a getter function
2. It resembles the `as_mut()` function that rust offers
3. It tells that the code may panic